lib/bolt/inventory.rb

```diff
@@ -73,31 +74,53 @@ def get_targets(targets)
       targets.map { |t| update_target(t) }
     end
 
-    # Should this be a public method?
     def config_for(node_name)
       data = @groups.data_for(node_name)
       if data
         Bolt::Util.symbolize_keys(data['config'])
       end
     end
 
+    def set_var(target, key, value)
+      data = { 'vars' => { key => value } }
+      set_vars_from_inv(target.name, data)
+    end
+
+    def vars(target)
+      @target_vars[target.name]
+    end
+
     #### PRIVATE ####
     #
     # For debugging only now
     def groups_in(node_name)
       @groups.data_for(node_name)['groups'] || {}
     end
 
+    def transform_data(data)
+      if data
+        data['config'] = Bolt::Util.symbolize_keys(data['config'])
+      end
+      data
+    end
+
     # Pass a target to get_targets for a public version of this
     # Should this reconfigure configured targets?
     def update_target(target)
-      inv_conf = config_for(target.name)
-      unless inv_conf
+      data = @groups.data_for(target.host)
+      processed = transform_data(data) || {}
+      unless processed['config']
         @logger.debug("Did not find #{target.name} in inventory")
-        inv_conf = {}
+        processed['config'] = {}
+      end
+
+      unless processed['vars']
+        @logger.debug("Did not find any variables for #{target.host} in inventory")
+        processed['vars'] = {}
       end
 
-      conf = Bolt::Util.deep_merge(@config.transport_conf, inv_conf)
+      set_vars_from_inv(target.name, processed) || {}
```

Comment:
Minor nit: seems clearer to pass `processed['vars']` and have `set_vars_from_inv` be closer to `set_vars_from_hash`.